### ISSUE TITLE
Cilium version 1.14 not work when set localRedirectPolicy=true and use CiliumLocalRedirectPolicy

### ISSUE THREAD

**Issue by hainamtvvt** [cilium/cilium#27499]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Cilium version 1.14 not work when set localRedirectPolicy=true and use CiliumLocalRedirectPolicy

My config 
```
SEED=$(head -c12 /dev/urandom | base64 -w0)
helm install idg-cilium cilium/cilium --version 1.14.0 \
    --namespace kube-system \
    --set tunnel=disabled \
    --set kubeProxyReplacement=true \
    --set bpf.lbMapMax=131072 \
    --set bpf.masquerade=true \
    --set enableIPv4Masquerade=true \
    --set bgpControlPlane.enabled=true \
    --set loadBalancer.algorithm=maglev \
    --set loadBalancer.mode=hybrid \
    --set maglev.tableSize=65521 \
    --set maglev.hashSeed=$SEED \
    --set ipam.mode=kubernetes \
    --set localRedirectPolicy=true \
    --set autoDirectNodeRoutes=true \
    --set hubble.enabled=true \
    --set hubble.relay.enabled=true \
    --set hubble.ui.enabled=true \
    --set operator.replicas=2 \
    --set hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}" \
    --set ipv4NativeRoutingCIDR="10.42.0.0/16" \
    --set k8sServiceHost=x.x.x.x \
    --set l2announcements.enabled=true \
    --set l2announcements.leaseDuration="3s" \
    --set l2announcements.leaseRenewDeadline="1s" \
    --set l2announcements.leaseRetryPeriod="500ms" \
    --set externalIPs.enabled=true \
    --set k8sServicePort=6443
```
And Deployment file CiliumLocalRedirectPolicy
```
apiVersion: "cilium.io/v2"
kind: CiliumLocalRedirectPolicy
metadata:
  name: "nodelocaldns"
  namespace: kube-system
spec:
  redirectFrontend:
    serviceMatcher:
      serviceName: rke2-coredns-rke2-coredns
      namespace: kube-system
  redirectBackend:
    localEndpointSelector:
      matchLabels:
        k8s-app: node-local-dns
    toPorts:
      - port: "53"
        name: udp-53
        protocol: UDP
      - port: "53"
        name: tcp-53
        protocol: TCP
```
**Cilium stuck ready 0/1**
![image](https://github.com/cilium/cilium/assets/63154819/16e13b52-83ad-4be9-bd68-55861622dee5)

### Cilium Version

Cilium version 1.14.0 

### Kernel Version

Linux master-01 5.4.0-125-generic

### Kubernetes Version

Kubernetes version 1.26

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ti-mo** [comment]
@hainamtvvt Please include some log output of those failing Cilium agent pods.

**Comment by hainamtvvt** [comment]
Hi @ti-mo  This issue has been addressed at Cilium version 1.14.1
